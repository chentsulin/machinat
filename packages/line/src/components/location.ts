/* eslint-disable import/prefer-default-export */
import {
  makeUnitSegment,
  UnitSegment,
  FunctionOf,
} from '@machinat/core/renderer';
import { annotateLineComponent } from '../utils';
import { LineComponent, LineMessageSegmentValue } from '../types';

/**
 * @category Props
 */
export type LocationProps = {
  title: string;
  address: string;
  latitude: number;
  longitude: number;
};

const __Location: FunctionOf<LineComponent<
  LocationProps,
  UnitSegment<LineMessageSegmentValue>
>> = function Location(node, path) {
  const { title, address, latitude, longitude } = node.props;

  return [
    makeUnitSegment(node, path, {
      type: 'location' as const,
      title,
      address,
      latitude,
      longitude,
    }),
  ];
};

/**
 * Location sends a user location message.
 * @category Component
 * @props {@link LocationProps}
 * @guides Check official [reference](https://developers.line.biz/en/reference/messaging-api/#location-message).
 */
export const Location: LineComponent<
  LocationProps,
  UnitSegment<LineMessageSegmentValue>
> = annotateLineComponent(__Location);
