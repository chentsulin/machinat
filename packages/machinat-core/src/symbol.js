// @flow
export const MACHINAT_ELEMENT_TYPE = Symbol('machinat.element');

export const MACHINAT_FRAGMENT_TYPE = Symbol('machinat.element.fragment');

export const MACHINAT_PAUSE_TYPE = Symbol('machinat.element.pause');

export const MACHINAT_PROVIDER_TYPE = Symbol('machinat.element.provider');

export const MACHINAT_THUNK_TYPE = Symbol('machinat.element.thunk');

export const MACHINAT_RAW_TYPE = Symbol('machinat.element.raw');

export const MACHINAT_NATIVE_TYPE = Symbol('machinat.component.native');

export const MACHINAT_SERVICES_CONTAINER = Symbol(
  'machinat.services.container'
);

export const MACHINAT_SERVICES_INTERFACE = Symbol(
  'machinat.services.interface'
);

export const MACHINAT_SERVICES_PROVIDER = Symbol('machinat.services.provider');
