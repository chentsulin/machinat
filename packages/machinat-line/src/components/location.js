/* eslint-disable import/prefer-default-export */
import { unitSegment } from '@machinat/core/renderer';
import { annotateLineComponent } from '../utils';

export const Location = (node, path) => {
  const { title, address, latitude, longitude } = node.props;

  return [
    unitSegment(node, path, {
      type: 'location',
      title,
      address,
      latitude,
      longitude,
    }),
  ];
};

annotateLineComponent(Location);
