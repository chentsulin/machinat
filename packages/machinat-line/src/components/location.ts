/* eslint-disable import/prefer-default-export */
import { unitSegment } from '@machinat/core/renderer';
import { annotateLineComponent } from '../utils';

const LineLocation = (node, path) => {
  const { title, address, latitude, longitude } = node.props;

  return [
    unitSegment(node, path, {
      type: 'location',
      title,
      address,
      latitude,
      longitude,
    }),
  ];
};

export const Location = annotateLineComponent(LineLocation);
