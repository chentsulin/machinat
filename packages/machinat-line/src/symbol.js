export const LINE_NAITVE_TYPE = Symbol.for('machinat.line.native');

export const API_PATH = Symbol.for('machinat.line.api_path');
export const HAS_BODY = Symbol.for('machinat.line.has_body');
export const THREAD_ID = Symbol.for('machinat.line.thread_id');

export const NO_BODY = Symbol.for('machinat.line.act.no_body');
