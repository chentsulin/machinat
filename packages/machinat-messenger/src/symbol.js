export const MESSENGER_NAITVE_TYPE = Symbol.for('machinat.messenger.native');
