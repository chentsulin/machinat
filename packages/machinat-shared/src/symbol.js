export const MACHINAT_ELEMENT_TYPE = Symbol.for('machinat.element');

export const MACHINAT_ROOT_ELEMENT = Symbol.for('machinat.element.root');

export const MACHINAT_FRAGMENT_TYPE = Symbol.for('machinat.element.fragment');

export const MACHINAT_ASYNC_TYPE = Symbol.for('machinat.element.async');
