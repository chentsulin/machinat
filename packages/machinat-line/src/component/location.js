/* eslint-disable import/prefer-default-export */
import { annotate, asNative, asUnit } from 'machinat-utility';

import { LINE_NAITVE_TYPE } from '../symbol';

export const Location = ({
  title,
  address,
  lat,
  latitude,
  long,
  longitude,
}) => [
  {
    type: 'location',
    title,
    address,
    latitude: latitude || lat,
    longitude: longitude || long,
  },
];

annotate(asNative(LINE_NAITVE_TYPE), asUnit(true))(Location);
