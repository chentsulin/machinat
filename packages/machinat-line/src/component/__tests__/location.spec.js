import Machinat from 'machinat';

import { Location } from '../location';

import { LINE_NAITVE_TYPE } from '../../symbol';

import render from './render';

it('is valid native unit component', () => {
  expect(typeof Location).toBe('function');

  expect(Location.$$native).toBe(LINE_NAITVE_TYPE);
  expect(Location.$$entry).toBe(undefined);
  expect(Location.$$unit).toBe(true);
});

it('render match snapshot', () => {
  expect(
    render(
      <Location
        title="WHERE AM I?"
        address="NARNIA"
        lat={51.756779}
        long={-1.189902}
      />
    ).map(act => act.value)
  ).toMatchSnapshot();
});
