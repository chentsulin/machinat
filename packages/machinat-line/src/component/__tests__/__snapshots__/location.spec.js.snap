// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render match snapshot 1`] = `
Array [
  Object {
    "address": "NARNIA",
    "latitude": 51.756779,
    "longitude": -1.189902,
    "title": "WHERE AM I?",
    "type": "location",
  },
]
`;
